0003-configuration-management-extension/README.md

```diff
@@ -0,0 +1,140 @@
+---
+name: Configuration Management Extension
+shortname: 3/CMX
+status: raw
+editor: Dmitry Sergeev <dsergeev@cybervisiontech.com>
+---
+
+## Introduction
+
+The Configuration Management Extension is a [Kaa Protocol](/0001-kaa-protocol/README.md) extension.
+
+It is intended to manage endpoint configuration distribution.
+
+## Requirements and constraints
+### Problems and possible solutions
+
+1. _Multiple configurations available for client._ It is possible that there will be more than one available configurations for endpoint (e.g., endpoint was offline while new configurations were added).
+   
+   Solution:
+   - Only latest configuration should be sent to an endpoint to reduce amount of traffic.
+
+2. _The server should know if a configuration has been delivered._ 
+
+   Solutions:
+   - Send PUBLISH message from client to "/status" topic. See below sections for details.
+
+## Use cases
+
+### UC1
+Configuration delivery by request. The endpoint should be able to receive latest configuration from CMX by request.
+
+### UC2
+Configuration delivery that is initiated by the server. The endpoint should receive latest configuration when it connects to a server if this configuration hadn't applied yet.
```

Comment:
Well, as I see now. There's few cases:
1) Endpoint is online and someone has uploaded new configuration on server. Server will push this configuration to the <endpoint_token>/push/json topic.
2) Endpoint if offline and someone has uploaded new configuration on server. Then endpoint should immediately receive configuration when it will restore connection with server.
3) Endpoint wants to get another configuration version (this is connected to another your comment, I agree that we should provide such functionality). In that case endpoint sends message to /pull/json topic, server responds to /push/json topic.